### PR TITLE
Update Icinga DB related docs

### PR DESCRIPTION
**Pull request by lippserd**
This PR introduces the following changes:

* The API section is promoted as almost every Icinga 2 setup requires the API
* Updated Icinga DB and Icinga DB Redis related instructions
* Deprecates the IDO
* Moves unnecessary steps/information from the installation chapter

### CODE REVIEW

**Comment by yhabteab** [comment]
**Ubuntu installation section**

Bildschirmfoto 2022-06-08 um 13 35 32

File: doc/02-installation.md [hunk 1]
```diff
@@ -450,31 +450,32 @@ yum install nagios-plugins-all
...
+or follow the steps below to set up the API quickly:
 
-```
-object ApiUser "icingaweb2" {
-  password = "Wijsn8Z9eRs5E25d"
-  permissions = [ "status/query", "actions/*", "objects/modify/*", "objects/query/*" ]
-}
+Run the following command to:
+
+* enables the `api` feature,
```
Comment (julianbrost) [inline doc/02-installation.md:447]: "enable" without an "s".

File: doc/09-object-types.md [hunk 2]
```diff
@@ -1425,6 +1425,11 @@ Configuration Attributes:
 
 ### IdoMySqlConnection <a id="objecttype-idomysqlconnection"></a>
 
+> **Note**
+>
+> This feature is DEPRECATED and may be removed in future releases.
+> Check the [roadmap](https://github.com/Icinga/icinga2/milestones).
```
Comment (julianbrost) [inline doc/09-object-types.md:1431]: To have the changes properly show up in icinga.com, this PR has to be backported to support/2.13, where I wouldn't deprecate the IDO yet but rather only in the 2.14 release.

One would either have to remember this when backporting the PR, or probably better as it removes the opportunity to forget it during backporting: make a separate PR for this change that will only go in 2.14.

File: doc/02-installation.md [hunk 3]
```diff
@@ -833,48 +833,6 @@ nano /etc/icinga2/conf.d/templates.conf
...
-  /usr/sbin/icinga2                   		| Shell wrapper for the Icinga 2 binary.
-  /usr/lib\*/icinga2				| Libraries and the Icinga 2 binary (use `find /usr -type f -name icinga2` to locate the binary path).
-  /usr/share/doc/icinga2              		| Documentation files that come with Icinga 2.
-  /usr/share/icinga2/include          		| The Icinga Template Library and plugin command configuration.
-  /var/lib/icinga2                    		| Icinga 2 state file, cluster log, master CA, node certificates and configuration files (cluster, api).
-  /var/run/icinga2                    		| PID file.
-  /var/run/icinga2/cmd                		| Command pipe and Livestatus socket.
-  /var/cache/icinga2                  		| status.dat/objects.cache, icinga2.debug files.
-  /var/spool/icinga2                  		| Used for performance data spool files.
-  /var/log/icinga2                    		| Log file location and compat/ directory for the CompatLogger feature.
```
Comment (julianbrost) [inline doc/02-installation.md:1189]: I wouldn't say this is completely useless, but probably better suited in some later more in-detail chapter.

File: doc/09-object-types.md [hunk 4]
```diff
@@ -1391,9 +1391,9 @@ Configuration Attributes:
   environment               | String                | **Optional.** Specify the Icinga environment. This overrides the `Environment` constant specified in the configuration or on the CLI with `--define`. Defaults to empty.
 
 
-### IcingaDB <a id="objecttype-icingadb"></a>
+### Icinga DB <a id="icinga-db"></a>
```
Comment (julianbrost) [inline doc/09-object-types.md:1394]: This would make this the only section that's not named exactly like the object type.

File: doc/09-object-types.md [hunk 5]
```diff
@@ -1409,10 +1409,10 @@ Configuration Attributes:
 
   Name                      | Type                  | Description
   --------------------------|-----------------------|----------------------------------
-  host                      | String                | **Optional.** Redis host for IcingaDB. Defaults to `127.0.0.1`.
-  port                      | Number                | **Optional.** Redis port for IcingaDB. Defaults to `6380`.
+  host                      | String                | **Optional.** Redis host. Defaults to `127.0.0.1`.
+  port                      | Number                | **Optional.** Redis port. Defaults to `6380` since the Redis server provided by the `icingadb-redis` package listens on that port.
   path                      | String                | **Optional.** Redix unix socket path. Can be used instead of `host` and `port` attributes.
```
Comment (julianbrost) [inline doc/09-object-types.md:1414]: When reworking this already, just fix that typo here as well?

File: doc/02-installation.md [hunk 6]
```diff
@@ -450,749 +428,210 @@ yum install nagios-plugins-all
...
-```
-
-Test it:
-
-```bash
-nano /etc/icinga2/conf.d/templates.conf
-```
-
-![Nano with syntax highlighting](images/installation/nano-syntax.png "Nano with Icinga 2 syntax highlighting")
+### Install Icinga DB Daemon <a id="install-icinga-db-daemon"></a>
```
Comment (julianbrost) [inline doc/02-installation.md:618]: This section shows up on the Raspbian/Fedora/OpenSUSE pages without the other Icinga DB sections.

**Comment by lippserd** [comment]
Since the doc branches are not merged yet, you need to manually use https://icinga.com/docs/icinga-db/example-config-and-docs/ for the Icinga DB doc links at the moment.

File: doc/09-object-types.md (hunk 2 above)
Comment (lippserd) [reply doc/09-object-types.md:1431]: I would immediately deprecate it as it is tied to the Icinga DB release and not an Icinga 2 version. The blog post will announce that as well.

File: doc/02-installation.md [hunk 7]
```diff
@@ -450,767 +428,240 @@ yum install nagios-plugins-all
...
-mysql_secure_installation
-```
-<!-- {% endif %} -->
+* Icinga 2 with its `icingadb` feature enabled,
+  responsible for publishing monitoring data to a Redis server, i.e. configuration and its runtime updates,
+  check results, state changes, downtimes, acknowledgements, notifications, and other events such as flapping
+* The [Icinga DB daemon](https://icinga.com/docs/icinga-db),
+  which synchronizes the data between the Redis server and a database
+* And Icinga Web with the
+  [Icinga DB Web](https://icinga.com/docs/icinga-db-web) module enabled,
```
[the remark this answers is not in the document]
Comment (julianbrost) [reply doc/02-installation.md:474]: Okay, same as https://github.com/Icinga/icingadb/pull/497#issuecomment-1162916331

File: doc/02-installation.md (hunk 3 above)
Comment (julianbrost) [reply doc/02-installation.md:1189]: So you really want this gone entirely (rather than moving it to `19-technical-concepts.md` for example)?

**Review by julianbrost — COMMENTED** [review]
Fine in general now, just that one comment remaining that got no reaction as far as I can see.

**Review by julianbrost — APPROVED** [review]
